plugins/com.google.cloud.tools.eclipse.swtbot/src/com/google/cloud/tools/eclipse/swtbot/SwtBotProjectActions.java

```diff
@@ -0,0 +1,313 @@
+    SwtBotMenuActions.openJavaPerpsective(bot);
+
+    // select the first project
+    bot.tree().setFocus();
+  }
+
+  public static void deleteProject(final SWTWorkbenchBot bot, final String projectName) {
+
+    SwtBotTestingUtilities.performAndWaitForWindowChange(bot, new Runnable() {
+      @Override
+      public void run() {
+        selectProject(bot, projectName).contextMenu("Delete").click();
+        // Wait for confirmation window to come up
+      }
+    });
+
+    // Select the "Delete project contents on disk (cannot be undone)"
+    bot.checkBox(0).click();
+
+    SwtBotTestingUtilities.clickButtonAndWaitForWindowChange(bot, bot.button("OK"));
+  }
+
+  /**
+   * Returns true if the specified project can be found in the 'Package Explorer' or 'Project View',
+   * otherwise returns false. Throws a WidgetNotFoundException exception if the 'Package Explorer'
+   * or 'Project Explorer' view cannot be found.
+   *
+   * @param bot The SWTWorkbenchBot
+   * @param projectName The name of the project to be found
+   * @return true if the project is found, and false if not found
+   */
+  public static boolean doesProjectExist(final SWTWorkbenchBot bot, String projectName) {
+    SWTBotView explorer = getExplorer(bot);
+
+    // Select the root of the project tree in the explorer view
+    Widget explorerWidget = explorer.getWidget();
+    Tree explorerTree = bot.widget(widgetOfType(Tree.class), explorerWidget);
+    SWTBotTreeItem[] allItems = new SWTBotTree(explorerTree).getAllItems();
+    for (int i = 0; i < allItems.length; i++) {
+      if (allItems[i].getText().equals(projectName)) {
+        return true;
+      }
+    }
+    return false;
+  }
+
+  /**
+   * Choose either the Package Explorer View or the Project Explorer view. Some perspectives have
+   * the Package Explorer View open by default, whereas others use the Project Explorer View.
+   * 
+   * @throws WidgetNoFoundException if an explorer is not found
+   */
+  public static SWTBotView getExplorer(final SWTWorkbenchBot bot) {
+    for (SWTBotView view : bot.views()) {
+      if (view.getTitle().equals("Package Explorer") || view.getTitle().equals("Project Explorer")) {
+        return view;
+      }
+    }
+    throw new WidgetNotFoundException(
+        "Could not find the 'Package Explorer' or 'Project Explorer' view.");
+  }
+
+  /**
+   * Returns the project root tree in Package Explorer.
+   */
+  public static SWTBotTree getProjectRootTree(SWTWorkbenchBot bot) {
+    SWTBotView explorer = getExplorer(bot);
+    Tree tree = bot.widget(widgetOfType(Tree.class), explorer.getWidget());
+    return new SWTBotTree(tree);
+  }
+
+  /**
+   * Returns true if there are errors in the Problem view. Returns false otherwise.
+   */
+  public static boolean hasErrorsInProblemsView(SWTWorkbenchBot bot) {
+    // Open Problems View by Window -> show view -> Problems
+    bot.menu("Window").menu("Show View").menu("Problems").click();
```

Comment:
There are other lines where we access a view. It could be useful to have a helper method that just wraps the call to `bot.menu("Window").menu("Show View").menu(<viewName>).click()`